### PR TITLE
Make LOAD DATA to respect tidb_dml_batch_size

### PR DESCRIPTION
**Pull request by alex-lx**
### What problem does this PR solve? 
#10524 

### What is changed and how it works?
Replace `loadDataInfo.SetMaxRowsInBatch(ctx.GetSessionVars().DMLBatchSize)` with
`loadDataInfo.SetMaxRowsInBatch(defaultLoadDataBatchCnt)`

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Comment by shenli** [comment]
@alex-lx Thanks!
Please fix the CI.

File: server/conn.go [hunk 1]
```diff
@@ -1076,8 +1078,12 @@ func (cc *clientConn) handleLoadData(ctx context.Context, loadDataInfo *executor
 
 	var shouldBreak bool
 	var prevData, curData []byte
-	// TODO: Make the loadDataRowCnt settable.
-	loadDataInfo.SetMaxRowsInBatch(defaultLoadDataBatchCnt)
+
+	if defaultLoadDataBatchCnt != initialDefaultLoadDataBatchCnt {
```
Comment (jackysp) [inline server/conn.go]: why not use `DMLBatchSize` directly?

Comment (alex-lx) [reply server/conn.go]: @jackysp https://github.com/pingcap/tidb/blob/master/server/tidb_test.go#L75 changed `defaultLoadDataBatchCnt` for testing purpose, so I keep `defaultLoadDataBatchCnt` a high priority to make the test happy.

**Comment by alex-lx** [comment]
@jackysp I'd love to add someone, however, I'm not so familiar in this codebase. In other words, would you like to show me where to add the test? 😟

**Comment by jackysp** [comment]
@alex-lx , maybe you could add some cases like,
https://github.com/pingcap/tidb/blob/d8589df2f321ce1f815cbd816846cea81c6c8111/executor/write_test.go#L1946

**Comment by shenli** [comment]
@alex-lx A friendly ping. Any update?

**Comment by alex-lx** [comment]
It's a little busy for me to complete this test code, so I'll close this PR and reopen it when it is ready.

**Pull request closed without merging**